seer_pas_sdk/core/unsupported.py

```diff
@@ -1448,3 +1448,363 @@ def _get_msdataindex_path(self, display_path: list):
             )
 
         return result
+
+    def get_search_data(
+        self,
+        analysis_id: str,
+        analyte_type: str,
+        rollup: str,
+        norm_method: str = "pepcal",
+    ):
+        """
+        Get analyte intensities data for a given PAS analysis.
+        Args:
+            analysis_id (str): ID of the analysis.
+            analyte_type (str): Type of the analyte. Must be either 'protein', 'peptide', precursor.
+            rollup (str): Intensities rollup method. Must be either 'np' or 'panel'.
+            norm_method (str): Search engine. Supported engines are: raw, engine, median, median80, pepcal. Default is 'pepcal'.
+
+        Returns:
+            pd.DataFrame: A dataframe with each row containing the analyte intensity measurement:
+                            'msrun_id', 'sample_id', 'nanoparticle' (if rollup is 'np'), 'protein_group', 'peptide' (for 'peptide' and 'precursor' analyte types), 'charge' (for 'precursor' analyte type),
+                            'intensity_log10', 'protein_group_q_value', 'q_value' (for 'precursor' analyte type), 'rt' and 'irt' (for 'peptide' and 'precursor' analyte types)
+        """
+        # 1. Get msrun data for analysis
+        samples = self.find_samples(analysis_id=analysis_id)
+        sample_name_to_id = {s["sample_name"]: s["id"] for s in samples}
+        # for np rollup, a row represents an msrun
+        msruns = self.find_msruns(sample_ids=sample_name_to_id.values())
+        file_to_msrun = {
+            os.path.basename(msrun["raw_file_path"]).split(".")[0]: msrun
+            for msrun in msruns
+        }
+        sample_to_msrun = {msrun["sample_id"]: msrun for msrun in msruns}
+
+        # for panel rollup, a row represents a sample
+
+        # 2. Get search results
+        # pull the np/panel file, or report.tsv for precursor mode
+        search_results = self.get_search_result(
+            analysis_id=analysis_id,
+            analyte_type=analyte_type,
+            rollup=rollup,
+        )
+        if analyte_type in ["protein", "peptide"]:
+            intensity_column = None
+            if norm_method == "raw":
+                intensity_column = (
+                    "Intensities Log10"
+                    if "Intensities Log10" in search_results.columns
+                    else "Intensity (Log10)"
+                )
+            elif norm_method == "engine":
+                intensity_column = (
+                    "DIA-NN Normalized Intensities Log10"
+                    if "DIA-NN Normalized Intensities Log10"
+                    in search_results.columns
+                    else "Normalized Intensity (Log10)"
+                )
+                if intensity_column not in search_results.columns:
+                    raise ValueError(
+                        "Error finding engine normalized intensities in search results. This is only supported for DIA-NN currently."
```

Comment:
Just for consistency with the other error messages:

```suggestion
                        "Engine normalized intensities not found in search results. This is only supported for DIA-NN currently."
```